astropy/units/tests/test_quantity_non_ufuncs.py

```diff
@@ -1280,11 +1280,33 @@ def test_nancumsum(self):
     def test_nanstd(self):
         self.check(np.nanstd)
 
+    def test_nanstd_out(self):
+        out = u.Quantity(-1)
+        o = np.nanstd(self.q, out=out)
+        assert o is out
+        assert o == np.nanstd(self.q)
+        # Also check array input, Quantity output.
+        o2 = np.nanstd(self.q.value, out=out)
+        assert o2 is out
+        assert o2.unit == u.dimensionless_unscaled
```

Comment:
this makes me wonder, shouldn't we also check the unit in `o/out` in the first half of the test ? (same question for `test_nanvar_out`)

Reply:
It really is a bit of overkill, since just checking equality checks the unit too (and I chose it to be initialized with the wrong one one purpose). Here, though, I felt that with an array input it made perhaps more sense.

But obviously feel free to add the unit check!

Reply:
> I chose it to be initialized with the wrong one one purpose

this wasn't obvious to me just from reading the test and I can easily imagine it looking like an oversight or a mistake to some future reader, so I expanded the tests with parametrization and added a comment to make this intention very explicit.

Reply:
OK, I think this covers the bases! Let's get it in.